File: Visitor/cpp-source/rooms/BathRoom.cpp
//
// Created by Pavel Akhtyamov on 2019-04-23.
//

#include "BathRoom.h"

void BathRoom::Accept(Visitor *visitor) {
  visitor->visit(this);
}
std::string BathRoom::GetName() {
  return "BathRoom";
}

File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/rooms/BathRoom.java
package ru.akhcheck.patterns.visitor.rooms;

import ru.akhcheck.patterns.visitor.workers.Visitor;

/** Санузел: единственная комната, где плитка идёт и на потолок. */
public class BathRoom extends ElementRoom {
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    @Override
    public String getName() {
        return "BathRoom";
    }
}
